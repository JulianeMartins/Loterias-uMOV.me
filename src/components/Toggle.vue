<template>
  <div
    class="pb-3 btn-group u-btn-group"
    role="group"
    aria-label="Basic radio toggle button group"
  >
    <template v-for="(item, index) in options" :key="index">
      <input
        @change="emitChange($event)"
        type="radio"
        class="btn-check"
        name="cardType"
        :id="item.value"
        autocomplete="off"
        v-model="selected"
        :value="item.value"
      />
      <label class="btn" :for="item.value">{{ item.label }}</label>
    </template>
  </div>
</template>
<script>
export default {
  name: "Toggle",
  props: ["options"],
  emits: ["change"],
  data() {
    return {
      selected: this.options[0].value,
    };
  },
  methods: {
    emitChange({ target }) {
      this.$emit("change", target.value);
    },
  },
};
</script>
