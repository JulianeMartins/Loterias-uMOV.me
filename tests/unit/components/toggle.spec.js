import { mount } from "@vue/test-utils";
import Toggle from "@/components/Toggle.vue";

const options = [
  {
    label: "Option 0",
    value: 0,
  },
  {
    label: "Option 1",
    value: 1,
  },
];

describe("Toggle", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(Toggle, {
      props: { options },
    });
  });
  test("match snapshot", () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test("set default option", () => {
    expect(wrapper.vm.selected).toBe(options[0].value);
  });
  [];
  test("emit change on change", () => {
    const label = wrapper.find('[datatestid = "label-0"]');
    label.trigger("change");
    expect(wrapper.emitted().change).toBeTruthy();
  });
});
