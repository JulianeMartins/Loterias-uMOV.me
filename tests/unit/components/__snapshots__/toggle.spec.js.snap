// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle match snapshot 1`] = `
<div
  aria-label="Basic radio toggle button group"
  class="pb-3 btn-group u-btn-group"
  role="group"
>
  
  
  <input
    autocomplete="off"
    class="btn-check"
    id="0"
    name="cardType"
    type="radio"
    value="0"
  />
  <label
    class="btn"
    datatestid="label-0"
    for="0"
  >
    Option 0
  </label>
  
  
  <input
    autocomplete="off"
    class="btn-check"
    id="1"
    name="cardType"
    type="radio"
    value="1"
  />
  <label
    class="btn"
    datatestid="label-1"
    for="1"
  >
    Option 1
  </label>
  
  
</div>
`;
